url93da129d40_marioskogias_blkin_hadoop_tgz-pJ8-mythrift_MyZipkinThriftJ8/src/mythrift/Span.java

```diff
@@ -0,0 +1,983 @@
+    if (field == null) {
+      throw new IllegalArgumentException();
+    }
+
+    switch (field) {
+    case TRACE_ID:
+      return isSetTrace_id();
+    case NAME:
+      return isSetName();
+    case ID:
+      return isSetId();
+    case PARENT_ID:
+      return isSetParent_id();
+    case ANNOTATIONS:
+      return isSetAnnotations();
+    case BINARY_ANNOTATIONS:
+      return isSetBinary_annotations();
+    }
+    throw new IllegalStateException();
+  }
+
+  @Override
+  public boolean equals(Object that) {
+    if (that == null)
+      return false;
+    if (that instanceof Span)
+      return this.equals((Span)that);
+    return false;
+  }
+
+  public boolean equals(Span that) {
+    if (that == null)
+      return false;
+
+    boolean this_present_trace_id = true;
+    boolean that_present_trace_id = true;
+    if (this_present_trace_id || that_present_trace_id) {
+      if (!(this_present_trace_id && that_present_trace_id))
+        return false;
+      if (this.trace_id != that.trace_id)
+        return false;
+    }
+
+    boolean this_present_name = true && this.isSetName();
+    boolean that_present_name = true && that.isSetName();
+    if (this_present_name || that_present_name) {
+      if (!(this_present_name && that_present_name))
+        return false;
+      if (!this.name.equals(that.name))
+        return false;
+    }
+
+    boolean this_present_id = true;
+    boolean that_present_id = true;
+    if (this_present_id || that_present_id) {
+      if (!(this_present_id && that_present_id))
+        return false;
+      if (this.id != that.id)
+        return false;
+    }
+
+    boolean this_present_parent_id = true && this.isSetParent_id();
+    boolean that_present_parent_id = true && that.isSetParent_id();
+    if (this_present_parent_id || that_present_parent_id) {
+      if (!(this_present_parent_id && that_present_parent_id))
+        return false;
+      if (this.parent_id != that.parent_id)
+        return false;
+    }
+
+    boolean this_present_annotations = true && this.isSetAnnotations();
+    boolean that_present_annotations = true && that.isSetAnnotations();
+    if (this_present_annotations || that_present_annotations) {
+      if (!(this_present_annotations && that_present_annotations))
+        return false;
+      if (!this.annotations.equals(that.annotations))
+        return false;
+    }
+
+    boolean this_present_binary_annotations = true && this.isSetBinary_annotations();
+    boolean that_present_binary_annotations = true && that.isSetBinary_annotations();
+    if (this_present_binary_annotations || that_present_binary_annotations) {
+      if (!(this_present_binary_annotations && that_present_binary_annotations))
+        return false;
+      if (!this.binary_annotations.equals(that.binary_annotations))
+        return false;
+    }
+
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    return 0;
+  }
+
+  @Override
+  public int compareTo(Span other) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 20. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.